File: frontend/lib/models/interest_model.dart
import 'package:frontend/models/user_model.dart';

class InterestModel {
  final String id;
  final String requestId;
  final String applicantId;
  final String status; // 'pending', 'accepted', 'rejected'
  final DateTime createdAt;
  final UserModel? applicant;

  InterestModel({
    required this.id,
    required this.requestId,
    required this.applicantId,
    required this.status,
    required this.createdAt,
    this.applicant,
  });

  factory InterestModel.fromJson(Map<String, dynamic> json) {
    return InterestModel(
      id: json['id'],
      requestId: json['request_id'],
      applicantId: json['applicant_id'],
      status: json['status'] ?? 'pending',
      createdAt: DateTime.parse(json['created_at']),
      applicant: json['users'] != null ? UserModel.fromJson(json['users']) : null,
    );
  }
}

File: backend/app/models/interest_model.py
# backend/app/models/interest_model.py

from datetime import datetime
from typing import Optional
from pydantic import BaseModel
import uuid
from .user_model import UserModel

class InterestModel(BaseModel):
    id: str = str(uuid.uuid4())
    request_id: str
    applicant_id: str
    status: str = 'pending'  # 'pending', 'accepted', 'rejected'
    created_at: datetime = datetime.now()
    applicant: Optional[UserModel] = None

    @classmethod
    def from_dict(cls, data: dict):
        if not data:
            return None
        
        applicant_data = data.get('users')
        applicant = UserModel.from_dict(applicant_data) if applicant_data else None
        
        return cls(
            id=data.get('id', str(uuid.uuid4())),
            request_id=data.get('request_id', ''),
            applicant_id=data.get('applicant_id', ''),
            status=data.get('status', 'pending'),
            created_at=datetime.fromisoformat(data.get('created_at', datetime.now().isoformat())),
            applicant=applicant
        )
    
    def to_dict(self) -> dict:
        return {
            'id': self.id,
            'request_id': self.request_id,
            'applicant_id': self.applicant_id,
            'status': self.status,
            'created_at': self.created_at.isoformat()
        }